### PR TITLE
Improve error handling on 'missing required columns'

### PR DESCRIPTION
**Pull request by rrchai**
Improve error message for 'missing required columns/wrong schema' type of errors.  

Not update preview table if it is because of 'missing columns'. 

related issue #133

### CODE REVIEW

File: app.R [hunk 1]
```diff
@@ -517,13 +527,14 @@ schema_to_display_lookup <- data.frame(schema_name, display_name)
...
-                            backgroundColor = styleEqual(errorDT$Value, rep("yellow", length(errorDT$Value) ) )) ## how to have multiple errors
-      })
+      if (update_preview) {
+        output$tbl <- DT::renderDT({
+          datatable(rawData(),
+                      options = list(lengthChange = FALSE, scrollX = TRUE)
+            ) %>% formatStyle(errorDT$Column,
+                              backgroundColor = styleEqual(errorDT$Value, rep("yellow", length(errorDT$Value) ) ))
+        })
+      }
```
Comment (sujaypatil96) [inline app.R:537]: Comment from @milen-sage: Highlight all columns in the preview table to "yellow". If it's visually appealing then retain it, else drop it. Also change the error message font color to "red".

**Review by sujaypatil96 — CHANGES_REQUESTED** [review]
Feel free to add commits to this PR, and then we can merge this PR.

File: app.R (hunk 1 above)
Comment (rrchai) [reply app.R]: update in e845dde: highlight all cells to "yellow" if there is a 'wrong schema' error

File: app.R [hunk 2]
```diff
@@ -482,6 +485,13 @@ schema_to_display_lookup <- data.frame(schema_name, display_name)
         errorDT <- data.frame(Column=sapply(annotation_status[inx_mt], function(i) i[[2]]),
                               Value=sapply(annotation_status[inx_mt], function(i) i[[4]][[1]]))
 
+      } else if (length(inx_ws) > 0) {  # wrong schema error(s): validating metadata miss any required columns
+        
+        type_error <- paste0("The submitted metadata does not contain all required column(s)")
+        help_msg <- "Please refer the correct template in the <b>Get Metadata Template</b> tab and
```
[the remark this answers is not in the document]
Comment (rrchai) [reply app.R]: update in fa443cb

File: app.R (hunk 1 above)
Comment (rrchai) [reply app.R]: I will make a seperate pr to change error message to red in

**Comment by sujaypatil96** [comment]
@rrchai: Can you avoid highlighting the columns in "yellow" as well? Just highlighting the contents in the table should suffice. Could you also change the waiter screen error message to "Wrong Schema" instead of showing the number of errors?

**Pull request merged**